src/providers/pkcs11_provider/key_metadata.rs

```diff
@@ -0,0 +1,115 @@
+// Copyright 2020 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+use super::{KeyInfo, KeyMetadata, Pkcs11Provider};
+use crate::key_info_managers;
+use crate::key_info_managers::KeyTriple;
+use log::{error, warn};
+use parsec_interface::operations::psa_key_attributes::*;
+use parsec_interface::requests::{ResponseStatus, Result};
+
+impl Pkcs11Provider {
+    /// Gets a key identifier and key attributes from the Key Info Manager.
+    pub(super) fn get_key_info(&self, key_triple: &KeyTriple) -> Result<([u8; 4], Attributes)> {
+        let store_handle = self.key_info_store.read().expect("Local ID lock poisoned");
```

Comment:
This module and these methods, not needing the mutex handle, make the code much clearer and easier to understand, thanks for that.
It means as well that we are reducing the scope of when the mutex is taken/freed which means that more parallelism will be available for the operations (which is good). I also believe that the error you saw on the other PR is a data race bug which is due to that.

Reply:
Yeah, probably, I'll have to look into it, but I have no clue how. The PKCS11 library should be thread-safe if we use a different session for each call (which we do)